libs/chatbot/lib/components/ChatBot/DeleteConversationModal.tsx

```diff
@@ -0,0 +1,87 @@
+import {
+  Alert,
+  Button,
+  Modal,
+  ModalBody,
+  ModalFooter,
+  ModalHeader,
+  Stack,
+  StackItem,
+} from '@patternfly-6/react-core';
+import * as React from 'react';
+import { getErrorMessage } from './helpers';
+
+type DeleteConversationModalProps = {
+  onClose: VoidFunction;
+  conversationId: string;
+  onDelete: () => Promise<unknown>;
+};
+
+const DeleteConversationModal = ({
+  onClose,
+  conversationId,
+  onDelete,
+}: DeleteConversationModalProps) => {
+  const [isDeleting, setIsDeleting] = React.useState(false);
+  const [error, setError] = React.useState<string>();
+
+  const handleDelete = React.useCallback(async () => {
+    setError(undefined);
+    setIsDeleting(true);
+    try {
+      await onDelete();
+      onClose();
+    } catch (e) {
+      setError(getErrorMessage(e));
+    } finally {
+      setIsDeleting(false);
+    }
+  }, [onDelete, onClose]);
+
+  return (
+    <Modal
+      isOpen
+      onClose={isDeleting ? undefined : onClose}
+      ouiaId="DeleteConversationModal"
+      aria-labelledby="delete-conversation-modal"
+      aria-describedby="modal-box-body-delete-conversation"
+      variant="small"
+    >
+      <ModalHeader
+        titleIconVariant="danger"
+        title="Delete conversation?"
+        labelId="delete-conversation-modal"
+      />
+      <ModalBody id="modal-box-body-delete-conversation">
+        <Stack hasGutter>
+          <StackItem>
+            Are you sure you want to delete conversation <b>{conversationId}</b>?
```

Comment:
Again, I think for the user the ID does not have any meaning. 
They only get to see the date so that's what I would show here again instead.